frontend/src/metabase/collections/containers/CollectionContent.jsx

```diff
@@ -123,8 +123,23 @@ function CollectionContent({ collection, collectionId, isAdmin, isRoot }) {
   };
 
   return (
-    <Search.ListLoader query={pinnedQuery} wrapped>
-      {({ list: pinnedItems }) => {
+    <Search.ListLoader
+      query={pinnedQuery}
+      loadingAndErrorWrapper={false}
+      keepPreviousList
+      wrapped
+    >
+      {({
```

Comment:
The current implementation heavily relies on clojures made by using a render prop pattern, so it won't be an easy refactor. Also, I'm sure @alxnddr had a good reason for that

Reply:
I had exactly the same reason when I worked on it :) However, I still introduced regressions there

Reply:
It would be definitely great to perform refactoring of this large thingy in a separate PR because the refactoring will generate more changes than the improvement Anton has done here

Reply:
Since we are pretty close to `0.40` release, it adds a risk of regressions that we might want to avoid

Reply:
Agree here 💪

Reply:
🤜❤️🤛